src/ApiCompat/Microsoft.DotNet.ApiCompatibility/ApiComparer.cs

```diff
@@ -47,53 +46,42 @@ public class ApiComparer : IApiComparer
             string? leftName = null,
             string? rightName = null)
         {
-            if (left == null)
-            {
-                throw new ArgumentNullException(nameof(left));
-            }
-
-            if (right == null)
-            {
-                throw new ArgumentNullException(nameof(right));
-            }
-
             AssemblyMapper mapper = new(GetComparingSettingsCore(leftName, rightName != null ? new[] { rightName } : null));
             mapper.AddElement(left, ElementSide.Left);
             mapper.AddElement(right, ElementSide.Right);
 
             DifferenceVisitor visitor = new();
             visitor.Visit(mapper);
-            return visitor.DiagnosticCollections.First();
+            return visitor.DiagnosticCollections[0];
```

Comment:
Avoiding Linq as the DiagnosticCollections are now exposed as a IReadOnlyList<T> which allows to index on it.